File: src/scripts/generateTeddyImages.js
import { generateTeddyImage, saveTeddyImage } from '../utils/imageGenerator';
import { supabase } from '../lib/supabase';
import { teddyData } from '../data/teddyData';

const generateAndSaveImages = async () => {
  for (const teddy of teddyData) {
    const imageUrl = await generateTeddyImage(teddy.name, teddy.description);
    if (imageUrl) {
      const savedUrl = await saveTeddyImage(teddy.name, imageUrl);
      if (savedUrl) {
        console.log(`Image for ${teddy.name} generated and saved: ${savedUrl}`);
        // Update the teddy data in Supabase with the new image URL
        const { error } = await supabase
          .from('terrible_teddies')
          .update({ imageUrl: savedUrl })
          .eq('name', teddy.name);
        
        if (error) {
          console.error(`Error updating image URL for ${teddy.name}:`, error);
        }
      }
    }
  }
};

generateAndSaveImages();

File: src/scripts/generateTeddyImages.py
import sys
import os

# Add the parent directory to the Python path
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from utils.imageGenerator import generateTeddyImages

def main():
    print("Generating teddy images...")
    images = generateTeddyImages()
    print(f"Generated {len(images)} teddy images.")

if __name__ == "__main__":
    main()